### DISCUSSION TITLE
Why it is so verbose in java?

### DISCUSSION THREAD

**Discussion by ericvondike** [graphql-java/graphql-java#2270] [General]
Why everything should be so verbose in JAVA? I mean just compare how we manage to do things with other languages like NodeJS, Go, Clojure, non of them needs so much code just to implement a provider. What is data fetcher? Why it is not a simple query? I do not understand why everything is so complex and verbose in java? Do we really need to write so much code when using graphql? or it is just a tutorial and the codes are there to make us understand how it works?

**Comment by lqc** [comment]
I recommend taking a look at these libraries:

- https://github.com/graphql-java-kickstart/graphql-java-tools 
- https://netflix.github.io/dgs/

Both are built on top of graphql-java and reduce the amount of boilerplate significantly. I think it's great that we have a library like this one that focuses on reliability and performance, so others can provide "sugar". That way, if you don't like the particular flavor, you can just make your own without having to redo the heavy lifting.

**Reply by phillip-kruger** [reply]
You can also look at microprofile graphql, and specifically the implementations https://github.com/eclipse/microprofile-graphql/#implementations

For example in Quarkus it's very easy to get started with GraphQL : https://quarkus.io/guides/microprofile-graphql

All of the implementations is build on graphql-java so if you need to get to the lower level code you can.

Cheers

**Reply by ericvondike** [reply]
Thank you @lqc and @phillip-kruger 

Your answers are really helpful.

Can I really use for example https://github.com/graphql-java-kickstart/graphql-java-tools 
in production?

I have some experience with it, but I thought it will not be a good choice for production.

**Comment by bbakerman** [comment]
Maybe you should check out KOTLIN!

Its based on the JVM (so a 20 year old runtime thats performant), it uses static typing (so that mistakes that should happen at build time do happen at build time) and has taken many of the lessons of Java over the last 20 years and made a concise language representation.

A data fetcher is the name chosen by this team to represent the callbacks that fields have.  I have heard them called "resolvers" in other implementations.

Our of interest - why would you turn up to a project called `graphql-java` (see those two words - `graphql` and `java`) and expect it it not to involve graphql and java deeply?

**Reply by gklijs** [reply]
He already knows Clojure, which is far better IMHO. Like lqc said there are several higher level libraries for Java.

**Comment by andimarek** [comment]
This forum is not for general Java discussion or for an explanation about why Java is what it is compared to Language X.

If you have specific questions about this library please ask them. "What is a DataFetcher" or "why is it named DataFetcher" are all valid question.

"Why everything should be so verbose in JAVA?" or "I do not understand why everything is so complex and verbose in java?" are not valid discussions for this topic.

Please start another discussion with more specific topics concerning this library. Thanks
